### PR TITLE
Add redirect async

### PR DESCRIPTION
**Pull request by george-zubrienko**
Closes https://github.com/SneaksAndData/adapta/issues/238

This PR adds final strokes for full asyncio-safe logging support with DataDog log handler by enabling async redirects from non-python processes. Thus we consider "multiple loggers" feature implemented - for async programs only, though. However since we aim at using asyncio-only programs, I consider this good enough :)

This change is required for Nexus to be compatible with frameworks that use earlier versions of adapta themselves.

## Scope

Implemented:
 - Added `redirect_async` method that is designed purely for async programs and uses an async context. Also added this to `LoggerInterface`
 - Added regular `redirect` context-managed method to `LoggerInterface`
 - Added support for `redirect` via sync and async loggers, with async logger pretending to be sync (backwards-compat option)

Additional changes:
- Added proper unit tests for output rediretion using libc printf
- Bumped `pylint` from 2 to 3
- Bumped minor from 2.7 to 2.8

## Checklist

- [x] GitHub issue exists for this change.
- [x] Unit tests added and they pass.
- [x] Pylint 10.0/10.0 without bloating `.pylintrc` with exceptions.
- [x] Self-review.

### CODE REVIEW

File: adapta/logs/_logger_interface.py [hunk 1]
```diff
@@ -72,3 +73,35 @@ def stop(self) -> None:
...
         Optional method to stop and flush the logger, if required.
         """
+
+    @contextmanager
+    @abstractmethod
+    def redirect(self, tags: Optional[Dict[str, str]] = None, **kwargs):
+        """
+         Redirects stdout to a temporary file and dumps its contents as INFO messages
+         once the wrapped code block finishes execution. Stdout is restored after the block completes execution.
+         Note that timestamps appended by the logger will not correlate with the actual timestamp
```
Comment (george-zubrienko) [inline adapta/logs/_logger_interface.py:83]:
```suggestion
```

File: adapta/logs/_logger_interface.py [hunk 2]
```diff
@@ -72,3 +73,35 @@ def stop(self) -> None:
...
+         of the reported message, if one is present in the output. This method works for the whole process,
+         including external libraries (C/C++ etc). Example usage:
+
+         with composite_logger.redirect():
```
Comment (george-zubrienko) [inline adapta/logs/_logger_interface.py:87]:
```suggestion
         with logger.redirect():
```

File: publish_hive.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (george-zubrienko) [inline publish_hive.json:1]: remove lol

**Comment by george-zubrienko** [comment]
Oh great tests fail on GH. Will investigate

**Review by s-vitaliy — APPROVED** [review]
LGTM

File: adapta/logs/_internal_logger.py [hunk 4]
```diff
@@ -1,6 +1,8 @@
 """
  Shared functionality for the MetadataLogger enricher implementations.
 """
+import ctypes
+
```
Comment (matt035343) [inline adapta/logs/_internal_logger.py:5]: One more rogue import

**Review by matt035343 — APPROVED** [review]

**Pull request merged**